drivers/gpio/gpio_mcux.c

```diff
@@ -328,10 +325,51 @@ static void gpio_mcux_port_isr(const struct device *dev)
 	/* Clear the gpio interrupts */
 	config->gpio_base->ISFR[0] = int_status;
 #endif  /* !(defined(FSL_FEATURE_PORT_HAS_NO_INTERRUPT) */
+	return int_status;
+}
+
+static void gpio_mcux_port_isr(const struct device *dev)
+{
+	const struct gpio_mcux_config *config = dev->config;
+	struct gpio_mcux_data *data = dev->data;
 
-	gpio_fire_callbacks(&data->callbacks, dev, int_status);
+	gpio_fire_callbacks(&data->callbacks, dev, read_gpio_interrupt_status(config));
 }
```

Comment:
I am unclear why this change was made to the gpio_mcux_port_isr. It looks like you took portions of the code and made it into a function that is called by gpio_mcux_port_isr

Reply:
it's to handle shared interrupts by multiple gpio controllers.

Reply:
actually, now that I look at this, I think it is a relic of an intermediate implementation while working on this, maybe it's not needed anymore

Reply:
Possibly, I think we can undo this change.